ocean_lib/models/test/test_factory_router.py

```diff
@@ -3,53 +3,140 @@
-    is_approved_token = factory_router.is_approved_token(
-        get_address_of_type(config, "Ocean")
+def test_opc_collector(config: Config, factory_router: FactoryRouter):
+    assert factory_router.get_opc_collector() == get_address_of_type(
+        config, "OPFCommunityFeeCollector"
     )
-    assert is_approved_token is True
 
 
 @pytest.mark.unit
-def test_mapping_ss_contracts(config, factory_router):
+def test_is_pool_template(config: Config, factory_router: FactoryRouter):
+    assert factory_router.is_pool_template(get_address_of_type(config, "poolTemplate"))
+
+
+# FactoryRouter methods
+
+
+@pytest.mark.unit
+def test_router_owner(factory_router: FactoryRouter):
+    assert Web3.isChecksumAddress(factory_router.router_owner())
+
+
+@pytest.mark.unit
+def test_factory(config: Config, factory_router: FactoryRouter):
+    assert factory_router.factory() == get_address_of_type(config, "ERC721Factory")
+
+
+@pytest.mark.unit
+def test_swap_ocean_fee(factory_router: FactoryRouter):
+    assert factory_router.swap_ocean_fee() == to_wei("0.001")
+
+
+@pytest.mark.unit
+def test_swap_non_ocean_fee(factory_router: FactoryRouter):
+    assert factory_router.swap_non_ocean_fee() == to_wei("0.002")
+
+
+@pytest.mark.unit
+def test_is_approved_token(config: Config, factory_router: FactoryRouter):
+    """Tests that Ocean token has been added to the mapping"""
+    assert factory_router.is_approved_token(get_address_of_type(config, "Ocean"))
+    assert not (factory_router.is_approved_token(ZERO_ADDRESS))
+
+
+@pytest.mark.unit
+def test_is_ss_contract(config: Config, factory_router: FactoryRouter):
     """Tests if ssContract address has been added to the mapping"""
-    assert factory_router.is_ss_contract(get_address_of_type(config, "Staking")) is True
+    assert factory_router.is_ss_contract(get_address_of_type(config, "Staking"))
 
 
 @pytest.mark.unit
-def test_fixed_rate_mapping(config, factory_router):
+def test_is_fixed_rate_contract(config: Config, factory_router: FactoryRouter):
     """Tests that fixedRateExchange address is added to the mapping"""
-    assert (
-        factory_router.is_fixed_rate_contract(get_address_of_type(config, "FixedPrice"))
-        is True
+    assert factory_router.is_fixed_rate_contract(
+        get_address_of_type(config, "FixedPrice")
+    )
+
+
+@pytest.mark.unit
+def test_is_dispenser_contract(config: Config, factory_router: FactoryRouter):
+    assert factory_router.is_dispenser_contract(
+        get_address_of_type(config, "Dispenser")
+    )
+
+
+@pytest.mark.unit
+def test_get_opc_fee(config: Config, factory_router: FactoryRouter):
+    assert factory_router.get_opc_fee(get_address_of_type(config, "Ocean")) == to_wei(
+        "0.001"
     )
+    assert factory_router.get_opc_fee(ZERO_ADDRESS) == to_wei("0.002")
+
+
+@pytest.mark.unit
+def test_get_opc_fees(factory_router: FactoryRouter):
+    assert factory_router.get_opc_fees() == [to_wei("0.001"), to_wei("0.002")]
```

Comment:
I thought the whole point of this PR was to make these single-entrypoint. Basically, we should have some constants in ocean.py where we set the fees, and if the fees change in the contracts, instead of adapting so many tests, we will just adapt the constant. So basically, in all tests, we should assert the values of those constants instead.

Reply:
That's a good idea.  I don't remember this being mentioned before and it's certainly not the whole point of this PR, but I'm happy to implement it.

Reply:
I added constants to the `test_factory_router.py` and refactored the tests to use them. I didn't add it to a `models/factory_router.py` because, in production, users should always get these values from the smart contracts. The constants are only used for tests.